File: bank-service/src/main/java/rs/edu/raf/repository/ExchangeAccountRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.ExchangeAccount;

import java.util.Optional;

@Repository
public interface ExchangeAccountRepository extends JpaRepository<ExchangeAccount, Long> {
    Optional<ExchangeAccount> findByCurrencyCode(String currency);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/ExchangeAccountRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.ExchangeAccount
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ExchangeAccountRepository : JpaRepository<ExchangeAccount, Long> {
    fun findByCurrencyCode(currency: String): Optional<ExchangeAccount>
}